services/provider/server/server.go

```diff
@@ -258,10 +259,85 @@ func (s *OCSProviderServer) GetStorageConfig(ctx context.Context, req *pb.Storag
 }
 
 // GetDesiredClientState RPC call to generate the desired state of the client
-func (s *OCSProviderServer) GetDesiredClientState(_ context.Context, _ *pb.GetDesiredClientStateRequest) (*pb.GetDesiredClientStateResponse, error) {
+func (s *OCSProviderServer) GetDesiredClientState(ctx context.Context, req *pb.GetDesiredClientStateRequest) (*pb.GetDesiredClientStateResponse, error) {
 
-	//TODO: Implement this in a different PR
-	return &pb.GetDesiredClientStateResponse{}, nil
+	// Get storage consumer resource using UUID
+	consumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, err
+	}
+
+	klog.Infof("Found storageConsumer for GetDesiredClientState")
+
+	// Verify Status
+	switch consumer.Status.State {
+	case ocsv1alpha1.StorageConsumerStateNotEnabled:
+		return nil, status.Errorf(codes.FailedPrecondition, "storageConsumer is in not enabled")
+	case ocsv1alpha1.StorageConsumerStateFailed:
+		return nil, status.Errorf(codes.Internal, "storageConsumer status failed")
+	case ocsv1alpha1.StorageConsumerStateConfiguring:
+		return nil, status.Errorf(codes.Unavailable, "waiting for the resources to be provisioned")
+	case ocsv1alpha1.StorageConsumerStateDeleting:
+		return nil, status.Errorf(codes.NotFound, "storageConsumer is in deleting phase")
+	case ocsv1alpha1.StorageConsumerStateReady:
+		kubeResources, err := s.getKubeResources(ctx, consumer)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to get kube resources. %v", err)
+		}
+
+		response := &pb.GetDesiredClientStateResponse{}
+
+		for i := range kubeResources {
+			kubeResource := kubeResources[i]
+			if kubeResource.GetObjectKind().GroupVersionKind().Group == "" ||
+				kubeResource.GetObjectKind().GroupVersionKind().Kind == "" ||
+				kubeResource.GetName() == "" {
```

Comment:
No reason to repeat the full access path to the GVK
```suggestion
			gvk := kubeResource.GetObjectKind().GroupVersionKind()
			if gvk.Group == "" || gvk.Kind == "" || kubeResource.GetName() == "" {
```